hs/src/Reach/Simulator/Core.hs

```diff
@@ -878,35 +916,35 @@ saferIndex n (_ : xs) = saferIndex (n -1) xs
 
 vUInt :: G.HasCallStack => DLVal -> Integer
 vUInt (V_UInt n) = n
-vUInt _ = impossible "unexpected error: expected integer value"
+vUInt b = impossible ("unexpected error: expected integer value: received " ++ show b)
```

Comment:
Haskell pointer: I recommend using `<>` over `++`, because it is more robust to changing the type, like if we wanted to use `Text` later

Reply:
thanks, i just always forget because `++` is so ubiquitous

Reply:
i can just fix this in the next PR, assuming you don't have an issue with `searchStores` once you understand what it's doing/why it's doing that